docs/user/openstack/install_upi.md

```diff
@@ -0,0 +1,861 @@
+# Installing OpenShift on OpenStack User-Provisioned Infrastructure
+
+The User-Provisioned Infrastructure (UPI) process installs OpenShift in stages, providing opportunities for modifications or integrating with existing infrastructure.
+
+It contrasts with the fully-automated Installer-Provisioned Infrastructure (IPI) which creates everything in one go.
+
+With UPI creating the cloud (OpenStack) resources (e.g. Nova servers, Neutron ports, security groups) is the responsibility of the person deploying OpenShift.
```

Comment:
I tend to read this line as
```
With UPI creating the cloud, resources [are] the responsibility...
```
instead of
```
With UPI, creating the cloud resources...
```

Consider adding a comma after `UPI`, for readability,.

Reply:
Commas have always been the Achilles' heel. Even in my native language \o/.